id_handling.c

```diff
@@ -183,47 +209,40 @@ esp_err_t ubirch_id_context_store(void) {
             current_id_context.short_name,
             current_id_context.state,
             current_id_context.memory_state.raw);
-    esp_err_t ret = ESP_FAIL;
-    if ((ret = store_value(current_id_context.short_name, KVKEY_PERSISTENT_STATE,
-                    &current_id_context.state,
-                    sizeof(current_id_context.state))) != ESP_OK) {
-        return ret;
-    }
-
-    if ((current_id_context.memory_state.uuid_updated == 1) &&
-            ((ret = store_value(current_id_context.short_name, KVKEY_UUID,
-                                UUID, sizeof(UUID))) != ESP_OK)) {
-        return ret;
-    }
-    current_id_context.memory_state.uuid_updated = 0;
-
-    if ((current_id_context.memory_state.password_updated == 1) &&
-            ((ret = store_value(current_id_context.short_name, KVKEY_PASSWORD,
-                                current_password, strlen(current_password))) != ESP_OK)) {
-        return ret;
+    esp_err_t ret = ESP_OK;
+
+    if (current_id_context.memory_state.id_blob_updated == 1) {
+        // create ID_BLOB from data
+        unsigned char id_blob_buffer[ID_BLOB_LENGTH];
+        unsigned char* id_blob_buffer_ptr = id_blob_buffer;
+
+        // copy state to
+        memcpy(id_blob_buffer_ptr, &current_id_context.state, sizeof(current_id_context.state));
+        id_blob_buffer_ptr = id_blob_buffer_ptr + sizeof(current_id_context.state);
+
+        // copy UUID to buffer
+        memcpy(id_blob_buffer_ptr, UUID, UUID_LEN);
+        id_blob_buffer_ptr = id_blob_buffer_ptr + UUID_LEN;
+
+        // copy password to buffer
+        memcpy(id_blob_buffer_ptr, current_password, PASSWORD_LENGTH + 1);
+        id_blob_buffer_ptr = id_blob_buffer_ptr + PASSWORD_LENGTH + 1;
+
+        // copy keypair to buffer
+        memcpy(id_blob_buffer_ptr, ed25519_secret_key, crypto_sign_SECRETKEYBYTES);
+        id_blob_buffer_ptr = id_blob_buffer_ptr + crypto_sign_SECRETKEYBYTES;
+
+        // copy next_key_update to buffer
+        memcpy(id_blob_buffer_ptr, &next_key_update, sizeof(time_t));
+        id_blob_buffer_ptr = id_blob_buffer_ptr + PASSWORD_LENGTH + 1;
```

Comment:
is this a copy/paste error? From my understanding, it should be:
id_blob_buffer_ptr = id_blob_buffer_ptr + sizeof(time_t)

Reply:
true, see comment above